test/cqlpy/test_materialized_view.py

```diff
@@ -1547,3 +1548,10 @@ def test_view_in_system_tables(cql, test_keyspace):
             assert view in res
             res = [ f'{r.table_name}.{r.index_name}' for r in cql.execute('select * from system."IndexInfo"')]
             assert view not in res
+
+# Test view representation in API
+def test_view_in_API(cql, test_keyspace):
+    with new_test_table(cql, test_keyspace, "p int PRIMARY KEY, v int") as base:
+        with new_materialized_view(cql, base, '*', 'v,p', 'v is not null and p is not null') as view:
+            res = rest_api.get_request(cql, f"/column_family/built_indexes/{base.replace('.',':')}")
+            assert view.split('.')[1] not in res
```

Comment:
I'm sorry, I really wanted to merge this version but I still don't understand this test. Basically you create a view and look for its name in some other irrelevant REST API request. What's the point? Don't we have a REST API with a list of built **views** that you can search for the view's name in?

Reply:
The point is that this REST API endpoint duplicates the system.IndexInfo table, so if we check that view is not in the table, we probably should do the same with the API endpoint.

Another justification for this is in how the endpoint is implemented. Here it is:

```c++
    cf::get_built_indexes.set(r, [&ctx, &sys_ks](std::unique_ptr<http::request> req) {
        return sys_ks.local().load_view_build_progress().then([ks, cf_name, &ctx](const std::vector<db::system_keyspace_view_build_progress>& vb) mutable {
            ...
            for (auto&& i : cf.get_index_manager().list_indexes()) {
                if (!vp.contains(secondary_index::index_table_name(i.metadata().name()))) {
                    res.emplace_back(i.metadata().name());
                }
            }
            ...
        });
```

So it really loads built _views_, then filters out those not listed in the index manager. If that breaks (e.g. #21445 touches that code) we'd better notice this early, don't we?

Reply:
Ok, I'll accept.

Still, doesn't it make sense to at least **also** use the REST API to list the views? Or maybe such a REST API doesn't exist at all?

All of this at least deserves a comment. This test doesn't really check how "views are represented in the REST API" in any reasonable sense of the word so you'd better explain what it does test.

Reply:
> Still, doesn't it make sense to at least also use the REST API to list the views? Or maybe such a REST API doesn't exist at all?

Hm, there's `/storage_service/view_build_statuses/{keyspace}/{view}`